prometheus-client/src/Prometheus/Metric/Summary.hs

```diff
@@ -79,7 +79,7 @@ summary info quantiles_ = Metric $ do
 
 instance Observer Summary where
     -- | Adds a new observation to a summary metric.
-    observe s v = doIO $ withMVar (reqSketch s) (`ReqSketch.insert` v)
+    observe s v = doIO $ withMVarMasked (reqSketch s) (\rs -> rs `ReqSketch.insert` v)
```

Comment:
**Suggestion:** Add a comment explaining why we are masking here. Something along the lines of your PR description would be great.

Reply:
+1 on adding a comment directly to the code.